src/runHandler.ts

```diff
@@ -35,19 +35,29 @@ export async function runHandler(
   )
     return
 
-  if (TestWatcher.isWatching() && watcher) {
-    watcher.runTests(request.include)
+  if (watchers.some(watcher => TestWatcher.isWatching(watcher.id)) && watchers.length > 0) {
+    watchers.forEach((watcher) => {
+      watcher.runTests(gatherTestItemsFromWorkspace(request.include ?? [], watcher.workspace.uri.fsPath))
+    })
+
     return
   }
 
-  const tests = request.include ?? gatherTestItems(ctrl.items)
-  const runner = new TestRunner(
-    vscode.workspace.workspaceFolders[0].uri.fsPath,
-    getVitestCommand(vscode.workspace.workspaceFolders[0].uri.fsPath),
-  )
-
   const run = ctrl.createTestRun(request)
-  await runTest(ctrl, runner, run, tests, 'run')
+
+  for (const folder of vscode.workspace.workspaceFolders) {
```

Comment:
The previous `await Promise.all` for runHandler did make sense to me. Because the running order doesn't matter when running tests in different workspaces, though it does when debugging

The previous suggested change was for `debugHandler`.